lib/compat.php

```diff
@@ -441,18 +441,17 @@ function gutenberg_inject_default_block_context( $args ) {
 			$registry = WP_Block_Type_Registry::get_instance();
 			$block_type = $registry->get_registered( $block->name );
 
-
 			// For WordPress versions that don't support the context API.
 			if ( ! $block->context ) {
 				$block->context = array();
 			}
 
 			// Inject the post context if not done by Core.
-			$needs_post_id = empty( $block_type->uses_context ) && in_array( 'postId', $block_type->uses_context, true );
+			$needs_post_id = ! empty( $block_type->uses_context ) && in_array( 'postId', $block_type->uses_context, true );
```

Comment:
@youknowriad Not sure what you intended this to be, but I'll assume it needs to be inverted to check emptiness before passing it to `in_array`.

Reply:
This change is causing some e2e tests to fail again. The current code in master seems equivalent to always setting `$needs_post_id` to false (the array can never be both empty AND have a key), so context is never injected. I pushed a commit to check if explicitly doing that fixes tests.